examples/matrix_multiplier/tests/test_matrix_multiplier.py

```diff
@@ -0,0 +1,181 @@
+            BinaryValue(
+                sum(
+                    [
+                        a_array[(i*A_COLUMNS_B_ROWS)+n] * b_array[(n*B_COLUMNS)+j]
+                        for n in range(A_COLUMNS_B_ROWS)
+                    ]
+                ),
+                n_bits=DATA_WIDTH*A_COLUMNS_B_ROWS,
+                bigEndian=False
+            )
+            for i in range(A_ROWS)
+            for j in range(B_COLUMNS)
+        ]
+
+        expected_output.append((a_array, b_array, c_expected))
+
+    monitor = MatMultMonitor(dut, True, callback=calc_c)
+    scoreboard = Scoreboard(dut)
+    scoreboard.add_interface(monitor, expected_output)
+
+    # Do multiplication
+    for A, B in zip(a_data(), b_data()):
+        yield RisingEdge(dut.clk)
+        for i, row in enumerate(A):
+            for j, value in enumerate(row):
+                dut.A[(i*A_COLUMNS_B_ROWS)+j] = value
+        for i, row in enumerate(B):
+            for j, value in enumerate(row):
+                dut.B[(i*B_COLUMNS)+j] = value
+        dut.i_valid = 1
+
+        yield RisingEdge(dut.clk)
+        dut.i_valid = 0
+
+    raise scoreboard.result
+
+@cocotb.coroutine
+def run_test(dut, a_data, b_data):
+    """For simulators that support multi-dimensional arrays.
+    """
+    cocotb.fork(Clock(dut.clk, 5, units='ns').start())
+
+    expected_output = []
+
+    def calc_c(transaction):
+        a_matrix, b_matrix, c_matrix = transaction
+
+        c_expected = [
+            [
+                BinaryValue(
+                    sum(
+                        [
+                            a_matrix[i][n] * b_matrix[n][j]
+                            for n in range(A_COLUMNS_B_ROWS)
+                        ]
+                    ),
+                    n_bits=DATA_WIDTH*A_COLUMNS_B_ROWS,
+                    bigEndian=False
+                )
+                for j in range(B_COLUMNS)
+            ]
+            for i in range(A_ROWS)
+        ]
+
+        expected_output.append((a_matrix, b_matrix, c_expected))
+
+    monitor = MatMultMonitor(dut, True, callback=calc_c)
+    scoreboard = Scoreboard(dut)
+    scoreboard.add_interface(monitor, expected_output)
+
+    # Do multiplication
+    for A, B in zip(a_data(), b_data()):
+        yield RisingEdge(dut.clk)
+        dut.A = A
+        dut.B = B
+        dut.i_valid = 1
+
+        yield RisingEdge(dut.clk)
+        dut.i_valid = 0
+
+    raise scoreboard.result
+
+def gen_random_matrix(rows, cols):
+    return [[getrandbits(DATA_WIDTH) for col in range(cols)] for row in range(rows)]
+
+def gen_a(num_samples=NUM_SAMPLES):
+    """Generate random matrix data for A
+    """
+    for i in range(num_samples):
+        yield gen_random_matrix(A_ROWS, A_COLUMNS_B_ROWS)
+
+def gen_b(num_samples=NUM_SAMPLES):
+    """Generate random matrix data for B
+    """
+    for i in range(num_samples):
+        yield gen_random_matrix(A_COLUMNS_B_ROWS, B_COLUMNS)
+
+
+if cocotb.SIM_NAME.lower().startswith(("icarus", "ghdl")):
```

Comment:
If this is a general limitation, we should add this to the Simulator Support section in the docs.